source/extensions/filters/network/zookeeper_proxy/filter.cc

```diff
@@ -216,13 +244,33 @@ void ZooKeeperFilter::onDecodeError() {
   setDynamicMetadata("opname", "error");
 }
 
-void ZooKeeperFilter::onRequestBytes(const uint64_t bytes) {
+void ZooKeeperFilter::onRequestBytes(const absl::optional<OpCodes> opcode, const uint64_t bytes) {
   config_->stats_.request_bytes_.add(bytes);
+
+  if (config_->enable_per_opcode_request_bytes_metrics_ && opcode.has_value()) {
+    if (*opcode == OpCodes::Connect) {
+      config_->stats_.connect_rq_bytes_.add(bytes);
+    } else {
+      ASSERT(config_->op_code_map_.contains(*opcode));
```

Comment:
What ensures that the `opcode` is always in the `op_code_map_`?
If it is certain, then please add a comment explaining why this invariant holds. Otherwise, add a if-else closure to handle the case where the op-code is unknown.

Reply:
Correct me if I'm wrong, but `deocdeOnData` is being called on the data-path (processing a user's request/response), and not only on the config-path (processing a config update).
Envoy does not allow exceptions on the data-path - if this throws an exception, that is not being caught, then the proxy will segfault.

I don't understand exactly what's going on here, but there might also be a case where a user will send a request containing a wrong op-code, and that will cause an exception to be thrown. If this is the case, then it could be an attack vector on the proxy.

Reply:
Is this going to be addressed in the follow up PR as well?

Reply:
Yes, I will try to fix it in the next PR.